### ISSUE TITLE
apiSource.securityDefinition changed behavior after upgrade from 1.0.0 to 1.0.2

### ISSUE THREAD

**Issue by cpoulsen-dezide** [gigaSproule/swagger-gradle-plugin#51]
Gradle 4.5.1

Having:
```
securityDefinition {
        json = '/securityDefinition.json'
}
```
Like mentioned various places in the README, has stopped working after upgrading to 1.0.2.
Previously it would successfully include the file from `src/main/resources/securityDefinition.json`, now the plugin fails with:
```
> Ambiguous method overloading for method com.fasterxml.jackson.databind.ObjectMapper#readTree.
  Cannot resolve which method to invoke for [null] due to overlapping prototypes between:
        [class com.fasterxml.jackson.core.JsonParser]
        [class [B]
        [class java.io.File]
        [class java.io.InputStream]
        [class java.io.Reader]
        [class java.lang.String]
        [class java.net.URL]
```

This is the same error as one gets if the plugin cannot locate the file specified.

Changing to:
```
securityDefinition {
        json = 'securityDefinition.json'
}
```
Gets things running again.

I also tried with absolute paths like mentioned in the README without any success (there is also an example with "${basedir}/..." which probably should be "${projectDir}/...") but none of the absolute paths are working (this may also have been the case pre 1.0.2 as I just tried it out now)

**Comment by gigaSproule** [comment]
The classpath changes caused this behaviour to change, but I would have thought that the absolute path would work. I will need to check it out when I have some spare time.

**Comment by gigaSproule** [comment]
I just tried the `spring-boot-jaxrs` sample with `jsonPath = "${project.getProjectDir()}/src/main/resources/securityDefinition.json"` and it worked fine. If I tried `jsonPath = "${project.getProjectDir()}/securityDefinition.json"` it correctly fails saying the file can't be found, printing out the expected path. `jsonPath` is for using an exact path and `json` is for a class path resource. Maybe we could join them together?

**Comment by cpoulsen-dezide** [comment]
Thank you for the quick reply!

I wasn't aware of the json vs. jsonPath until now. The README updates definitely looks like an improvement. 

I know there is a lot of flexibility to gradle so it can be hard to document all ways to achieve something, but the comment about getClass().getResourceAsStream below the securityDefinition.json example makes me no wiser to where to put the file (I had to experiment initially to find a location that worked).

You could consider only supporting the jsonPath option? - "${projectDir}/src/main/...." clearly expresses where the file is located, IMO.

**Comment by gigaSproule** [comment]
I was thinking of merging them into 1. This was originally just a wrapper around [swagger-maven-plugin](https://github.com/kongchen/swagger-maven-plugin), so it still has the same configurations. Now that it has been re-written, it might make sense to move away from their config setup, checking if the file exists on the classpath and then using the path if it's not on the classpath (e.g. try and find "/securityDefinition.json" in the classpath, then trying to find just "/securityDefinition.json")

**Comment by cpoulsen-dezide** [comment]
I think that it would be a good default to only look for the file in the filesystem. If the definition lives in an external module / jar one can extract it prior to using the task. (alternatively support prefixes like "classpath:<filename>" and "file:<filename>" although i think that most projects will have the definitions directly in the filesystem) - but I think I'm in favor of just looking in the filesystem.

**Comment by gigaSproule** [comment]
To be fixed as part of issue #43

**Comment by gigaSproule** [comment]
Closing this so there is only 1 issue tracking the changes for security definitions rather than 2.
